core/domain/base_model_validators.py

```diff
@@ -97,9 +98,19 @@ def __init__(self, field_name, model_class, model_ids):
                 'When fetching instances of UserSettingsModel, please use ' +
                 'UserSettingsModelFetcherDetails instead of ' +
                 'ExternalModelFetcherDetails')
+        filtered_model_ids = []
+        for model_id in model_ids:
+            if not re.compile(
+                    '^[A-Za-z0-9-_]{1,%s}$' % base_models.ID_LENGTH).match(
+                        model_id):
+                raise Exception(
+                    'The model id %s in the field \'%s\' '
+                    'is invalid' % (model_id, field_name))
+            else:
+                filtered_model_ids.append(model_id)
```

Comment:
I would do just a basic check that the ID is not `''` or `None` here since we cannot make assumptions about the ID format.

Reply:
Done.